{
    "type": "error",
    "data": "Error executing verb \"cluster_graph\" in create_base_entity_graph: EmptyNetworkError",
    "stack": "Traceback (most recent call last):\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\datashaper\\workflow\\workflow.py\", line 410, in _execute_verb\n    result = node.verb.func(**verb_args)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graphrag\\index\\verbs\\graph\\clustering\\cluster_graph.py\", line 61, in cluster_graph\n    results = output_df[column].apply(lambda graph: run_layout(strategy, graph))\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\pandas\\core\\series.py\", line 4924, in apply\n    ).apply()\n      ^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\pandas\\core\\apply.py\", line 1427, in apply\n    return self.apply_standard()\n           ^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\pandas\\core\\apply.py\", line 1507, in apply_standard\n    mapped = obj._map_values(\n             ^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\pandas\\core\\base.py\", line 921, in _map_values\n    return algorithms.map_array(arr, mapper, na_action=na_action, convert=convert)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\pandas\\core\\algorithms.py\", line 1743, in map_array\n    return lib.map_infer(values, mapper, convert=convert)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"lib.pyx\", line 2972, in pandas._libs.lib.map_infer\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graphrag\\index\\verbs\\graph\\clustering\\cluster_graph.py\", line 61, in <lambda>\n    results = output_df[column].apply(lambda graph: run_layout(strategy, graph))\n                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graphrag\\index\\verbs\\graph\\clustering\\cluster_graph.py\", line 171, in run_layout\n    clusters = run_leiden(graph, strategy)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graphrag\\index\\verbs\\graph\\clustering\\strategies\\leiden.py\", line 26, in run\n    node_id_to_community_map = _compute_leiden_communities(\n                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graphrag\\index\\verbs\\graph\\clustering\\strategies\\leiden.py\", line 61, in _compute_leiden_communities\n    community_mapping = hierarchical_leiden(\n                        ^^^^^^^^^^^^^^^^^^^^\n  File \"<@beartype(graspologic.partition.leiden.hierarchical_leiden) at 0x1cf07342980>\", line 304, in hierarchical_leiden\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graspologic\\partition\\leiden.py\", line 588, in hierarchical_leiden\n    hierarchical_clusters_native = gn.hierarchical_leiden(\n                                   ^^^^^^^^^^^^^^^^^^^^^^^\nleiden.EmptyNetworkError: EmptyNetworkError\n",
    "source": "EmptyNetworkError",
    "details": null
}
{
    "type": "error",
    "data": "Error running pipeline!",
    "stack": "Traceback (most recent call last):\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graphrag\\index\\run\\run.py\", line 227, in run_pipeline\n    result = await _process_workflow(\n             ^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graphrag\\index\\run\\workflow.py\", line 91, in _process_workflow\n    result = await workflow.run(context, callbacks)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\datashaper\\workflow\\workflow.py\", line 369, in run\n    timing = await self._execute_verb(node, context, callbacks)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\datashaper\\workflow\\workflow.py\", line 410, in _execute_verb\n    result = node.verb.func(**verb_args)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graphrag\\index\\verbs\\graph\\clustering\\cluster_graph.py\", line 61, in cluster_graph\n    results = output_df[column].apply(lambda graph: run_layout(strategy, graph))\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\pandas\\core\\series.py\", line 4924, in apply\n    ).apply()\n      ^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\pandas\\core\\apply.py\", line 1427, in apply\n    return self.apply_standard()\n           ^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\pandas\\core\\apply.py\", line 1507, in apply_standard\n    mapped = obj._map_values(\n             ^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\pandas\\core\\base.py\", line 921, in _map_values\n    return algorithms.map_array(arr, mapper, na_action=na_action, convert=convert)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\pandas\\core\\algorithms.py\", line 1743, in map_array\n    return lib.map_infer(values, mapper, convert=convert)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"lib.pyx\", line 2972, in pandas._libs.lib.map_infer\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graphrag\\index\\verbs\\graph\\clustering\\cluster_graph.py\", line 61, in <lambda>\n    results = output_df[column].apply(lambda graph: run_layout(strategy, graph))\n                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graphrag\\index\\verbs\\graph\\clustering\\cluster_graph.py\", line 171, in run_layout\n    clusters = run_leiden(graph, strategy)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graphrag\\index\\verbs\\graph\\clustering\\strategies\\leiden.py\", line 26, in run\n    node_id_to_community_map = _compute_leiden_communities(\n                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graphrag\\index\\verbs\\graph\\clustering\\strategies\\leiden.py\", line 61, in _compute_leiden_communities\n    community_mapping = hierarchical_leiden(\n                        ^^^^^^^^^^^^^^^^^^^^\n  File \"<@beartype(graspologic.partition.leiden.hierarchical_leiden) at 0x1cf07342980>\", line 304, in hierarchical_leiden\n  File \"D:\\Anaconda\\envs\\gr\\Lib\\site-packages\\graspologic\\partition\\leiden.py\", line 588, in hierarchical_leiden\n    hierarchical_clusters_native = gn.hierarchical_leiden(\n                                   ^^^^^^^^^^^^^^^^^^^^^^^\nleiden.EmptyNetworkError: EmptyNetworkError\n",
    "source": "EmptyNetworkError",
    "details": null
}
